src/network/gossip_test.rs

```diff
@@ -94,3 +130,72 @@ async fn test_gossip_communication() {
     }
     assert_eq!(receive_counts, 1);
 }
+
+#[tokio::test]
+#[serial]
+async fn test_peer_discovery() {
+    // Create two keypairs for our test nodes
+    let keypair1 = Keypair::generate();
+    let keypair2 = Keypair::generate();
+    let keypair3 = Keypair::generate();
+
+    // Create configs with different ports
+    let node1_addr = format!("/ip4/{HOST_FOR_TEST}/udp/{BASE_PORT_FOR_TEST}/quic-v1");
+    let node2_port = BASE_PORT_FOR_TEST + 1;
+    let node2_addr = format!("/ip4/{HOST_FOR_TEST}/udp/{node2_port}/quic-v1");
+    let node3_port = BASE_PORT_FOR_TEST + 2;
+    let node3_addr = format!("/ip4/{HOST_FOR_TEST}/udp/{node3_port}/quic-v1");
+
+    let config1 = Config::new(node1_addr.clone(), node2_addr.clone())
+        .with_contact_info_advertisement_interval(Duration::from_millis(100));
+    // Nodes 2 and 3 are not connected to one another via bootstrap
+    let config2 = Config::new(node2_addr.clone(), node1_addr.clone())
+        .with_contact_info_advertisement_interval(Duration::from_millis(100));
+    let config3 = Config::new(node3_addr.clone(), node1_addr.clone())
+        .with_contact_info_advertisement_interval(Duration::from_millis(100));
+
+    // Create channels for system messages
+    let (system_tx1, _) = mpsc::channel::<SystemMessage>(100);
+    let (system_tx2, _) = mpsc::channel::<SystemMessage>(100);
+    let (system_tx3, _) = mpsc::channel::<SystemMessage>(100);
+
+    // Create gossip instances
+    let mut gossip1 = SnapchainGossip::create(
+        keypair1.clone(),
+        &config1,
+        system_tx1,
+        false,
+        FarcasterNetwork::Devnet,
+    )
+    .unwrap();
+    let mut gossip2 = SnapchainGossip::create(
+        keypair2.clone(),
+        &config2,
+        system_tx2,
+        false,
+        FarcasterNetwork::Devnet,
+    )
+    .unwrap();
+
+    let mut gossip3 = SnapchainGossip::create(
+        keypair3.clone(),
+        &config3,
+        system_tx3,
+        false,
+        FarcasterNetwork::Devnet,
+    )
+    .unwrap();
+
+    // Spawn gossip tasks
+    tokio::spawn(async move {
+        gossip1.start().await;
+    });
+    tokio::spawn(async move {
+        gossip2.start().await;
+    });
+    tokio::spawn(async move {
+        gossip3.start().await;
+    });
+
+    tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
No asserts?

Reply:
Ah good catch, this was a draft function that I wrote. I ended up incorporating the behavior I wanted to test into the existing test.